### DISCUSSION TITLE
Opentelemetry instrumentation for google pubsub.

### DISCUSSION THREAD

**Discussion by OmarEltamasehy** [open-telemetry/opentelemetry-go#2221] [Q&A]
Hi,
I hope all are doing well,
Appreciate your support for providing me with docs or link to do cloud pubsub tracing if opentelemetry instrumentation pubsub already supported from your side.

Thanks in advance.

**Comment by srikanthccv** [comment]
There is no instrumentation for that in [python-contrib](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation) repo. GCP or community maintained instrumentation might exist outside this org. @aabmass

**Reply by aabmass** [reply]
@lonewolf3739 is correct, the plan is to instrument these in the Google PubSub client libraries. It is done in the NodeJS client already, but needs to be implemented in the other client libraries. I'd recommend letting the PubSub folks know you're interested in this feature.
